### PR TITLE
Add clear option to date pickers in Project Phases and Milestones Tables

### PR DESCRIPTION
**Pull request by jgabitto**
- Closes https://github.com/cityofaustin/atd-data-tech/issues/6159
- Revise conditional logic in Project Milestones table to clear date
- Add clear icon button to clear date in date pickers

How to view changes
- Go to netlify preview and sign into moped
- Go to projects pages and select an existing project
- Go to timeline tab and add phase and milestone if none are present
- Once they are added, click on edit button on the right and change values in date pickers
- Click the clear icon button
- Click on the check mark on the right

### CODE REVIEW

**Review by amenity — APPROVED** [review]
@jgabitto this is perfect!! Works beautifully. Thank you so much. 🙏

File: moped-editor/src/views/projects/projectView/ProjectTimeline.js [hunk 1]
```diff
@@ -384,6 +406,7 @@ const ProjectTimeline = ({ refetch: refetchSummary }) => {
       editComponent: props => (
         <DateFieldEditComponent
           {...props}
+          tableType="Milestone"
```
Comment (chiaberry) [inline moped-editor/src/views/projects/projectView/ProjectTimeline.js:409]: what does `tableType` do?

Comment (jgabitto) [reply moped-editor/src/views/projects/projectView/ProjectTimeline.js:409]: Hey Chia thanks for catching that! `tableType ` was a prop I was using in one of my attempts in solving the problem. I'm going to create a new commit that removes that.

**Comment by jgabitto** [comment]
@mateoclarke @sergiogcx @amenity @chiaberry So the files `NewProjectMap.js`, `NewProjectView.js` and `ProjectTeamTable.js` have been modified due to my code formatter Prettier. Not sure if I should undo those changes or not as they are not relevant to the issue.

**Review by mateoclarke — APPROVED** [review]
mason_ramsey

**Review by sergiogcx — APPROVED** [review]
It works! 🚢 🚀

**Comment by sergiogcx** [comment]
@jgabitto and yes, those style changes are fine! 👍

**Pull request merged**